crates/phala-node-rpc-ext/src/lib.rs

```diff
@@ -52,17 +62,15 @@ pub trait NodeRpcExtApi<BlockHash> {
 struct NodeRpcExt<BE, Block: BlockT, Client, P> {
     client: Arc<Client>,
     backend: Arc<BE>,
-    is_archive_mode: bool,
     pool: Arc<P>,
     _phantom: PhantomData<Block>,
 }
 
 impl<BE, Block: BlockT, Client, P> NodeRpcExt<BE, Block, Client, P> {
-    fn new(client: Arc<Client>, backend: Arc<BE>, is_archive_mode: bool, pool: Arc<P>) -> Self {
+    fn new(client: Arc<Client>, backend: Arc<BE>, _is_archive_mode: bool, pool: Arc<P>) -> Self {
         Self {
             client,
             backend,
-            is_archive_mode,
```

Comment:
So in the new version, pruned node also supports this RPC. Will it give error like "state discarded" if it hits the pruned block state?

Reply:
> Will it give error like "state discarded" if it hits the pruned block state?

Yes, it will.